src/App.tsx

```diff
@@ -1,25 +1,45 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
 import { getNumbers } from './utils';
+import { Pagination } from './components/Pagination';
 
 // eslint-disable-next-line @typescript-eslint/no-unused-vars
 const items = getNumbers(1, 42).map(n => `Item ${n}`);
 
 export const App: React.FC = () => {
+  const [perPage, setPerPage] = useState(5);
+  const [currentPage, setCurrentPage] = useState(1);
+  const totalNumber: number = items.length;
+  const start: number = currentPage * perPage - perPage;
+  const end: number =
+    currentPage * perPage > totalNumber ? totalNumber : currentPage * perPage;
+  const visibleItems: string[] = items.slice(start, end);
+
+  const handlePerPageItems = (e: React.ChangeEvent<HTMLSelectElement>) => {
+    const value = e.target.value;
+
+    setPerPage(+value);
+    setCurrentPage(1);
+  };
+
+
   return (
     <div className="container">
       <h1>Items with Pagination</h1>
 
       <p className="lead" data-cy="info">
-        Page 1 (items 1 - 5 of 42)
+         Page {currentPage} (items {start + 1} - {end} of {totalNumber})
```

Comment:
The pagination info format is correct. Ensure that the values for `start`, `end`, and `totalNumber` are calculated correctly to display accurate information.